src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -149,19 +161,91 @@ public int getTotalMessageCount(){
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
   
-  /* Returns all laptops */
-  public List<Laptop> getAllLaptops(){
-	 List<Laptop> laptops = new ArrayList<>();
-	 laptops.add(new Laptop("Dell","Black","Windows",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",13,1200.00));
-	 laptops.add(new Laptop("Dell","Grey","Windows",15,800.00));
-	 laptops.add(new Laptop("Dell","Black","Linux",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",15,1699.00));
-	 laptops.add(new Laptop("Macbook Pro","Black","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Black","Mac",15,1900.00));
-	 laptops.add(new Laptop("Macbook Pro","Grey","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Grey","Mac",15,1900.00));
-
-	 return laptops;
-  }
-}
\ No newline at end of file
+  /**
+   * @return list of all Laptops
+   */
+  public List<Laptop> getAlllaptopList(){
+	 return laptopList;
+  }
+  
+  /** Adds a laptop to the LaptopList
+   * @param Laptop to add to the list 
+   */
+  public void addLaptop(Laptop newLaptop) {
+	  laptopList.add(newLaptop);
+  }
+  
+  /**Returns a laptop list with laptops of a certain brand*/
+  public List<Laptop> searchBrand(String brand){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> brandreturn=new ArrayList<>();
+	  for(Laptop e:laptopList) {
+		  if(e.getBrand().equals(brand.toLowerCase()))
+			  brandreturn.add(e);
+	  }
+	  return brandreturn; 
+  }
+  
+  /**Returns a laptop list with laptops of a certain OS*/
+  public List<Laptop> searchOs(String os){
+	  List<Laptop> osreturn=new ArrayList<>();
+	  for(Laptop e:laptopList) {
+		  if(e.getOS().equals(os.toLowerCase()))
+			  osreturn.add(e);
+	  }
+	  return osreturn; 
+  }
+  
+  /**Returns a laptop list with laptops of a certain color*/
+  public List<Laptop> searchColor(String color){
+	  List<Laptop> colorreturn=new ArrayList<>();
+	  for(Laptop e:laptopList) {
+		  if(e.getColor().equals(color.toLowerCase()))
```

Comment:
Is it possible for any of the values you're using on this line (and lines similar to it below) to be null?